docs/main/source/widget_renderer.rst

```diff
@@ -0,0 +1,134 @@
+
+Incorporating **events** and **callback functions** into the `widgetRenderer` enhances the interactivity of the simulation. Users can tie actions to simulation steps, key presses, or other event triggers to update UI elements or adjust simulation parameters in real time.
+
+Key Concepts
+^^^^^^^^^^^^
+
+1. **Callback Functions**: Custom functions that are executed in response to a specific event (such as a button press).
+2. **Event System**: Tissue Forge provides built-in events that allow custom behaviors to be triggered by particle interactions, or user inputs.
+
+Examples
+^^^^^^^^
+
+Adding Widgets
+--------------
+You can add various widgets, such as buttons, input fields, and display outputs, to the simulation interface. These widgets allow users to interact with the simulation in real time by modifying simulation parameters or visualizing ongoing metrics.
+
+- **Displaying Simulation Metrics**
+
+  To visualize key metrics like simulation time or particle count:
+
+  .. code-block:: python
+
+      tf.system.show_widget_time()
+      tf.system.show_widget_particle_number()
+      tf.system.show_widget_bond_number()
+
+These methods enable dynamic tracking of the simulation's state directly within the UI.
+
+Custom Widget Outputs
+^^^^^^^^^^^^^^^^^^^^^
+
+You can add custom widgets to display dynamic values, such as the current standard deviation of a random force in the simulation. 
+
+First, create a random force applied to particles in the simulation:
+
+.. code-block:: python
+
+    # Create a random force with mean 0 and standard deviation 50
+    rforce = tf.Force.random(mean=0, std=50)
+
+Next, add a widget to display the current standard deviation (`std`) of the random force:
+
+.. code-block:: python
+
+    # Add a widget to display the noise level
+    idx_noise = tf.system.add_widget_output_float(rforce.std, 'Noise')
+
+.. note::
+   The `add_widget_output_float` function requires a float value. Ensure that the variable passed (`rforce.std` in this case) is of the correct type.
+
+To ensure that the displayed noise level updates during the simulation, a **callback function** is necessary.
+
+Callback: Update Displayed Values
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+You can register a callback function to update the widget display as the simulation progresses:
+
+.. code-block:: python
+
+    def update_display_noise():
+        tf.system.set_widget_output_float(idx_noise, rforce.std)
+
+Register this callback to be called at each simulation step:
+
+.. code-block:: python
+
+    # Register the callback to update the noise display on each simulation step
+    tf.event.on_simulation_step(update_display_noise)
```

Comment:
This function doesn't exist (though it should). An appropriate replacement would be `tf.event.on_time(period=0, invoke_method=lambda e: update_display_noise())`.